### PR TITLE
Ia brendannorth

### PR DESCRIPTION
**Pull request by ianyimi**
unfortunately this also has the balloonski shop changes I made as well bc i based this branch off that one

### CODE REVIEW

File: src/themes/components/Link.tsx [hunk 1]
```diff
@@ -10,10 +10,15 @@ type LinkProps = {
...
   color?: string;
+  blank?: boolean;
 };
 
 const Link = (props: LinkProps) => {
-  const { audio, link, position, rotY = 0, size = 1.6, src } = props;
+  const { audio, link, position, rotY = 0, size = 1.6, src, blank } = props;
+
+  if (blank) {
+    return <></>;
```
Comment (alex-shortt) [inline src/themes/components/Link.tsx:20]: don't need blank prop

File: src/themes/components/Artwork.tsx [hunk 2]
```diff
@@ -27,6 +28,7 @@ const Artwork = (props: ArtworkProps) => {
           link={piece.link}
           size={piece.size}
           src={piece.src}
+          blank={piece.blank}
```
Comment (alex-shortt) [inline src/themes/components/Artwork.tsx:31]: why not return 
```
!piece.blank &&  <Link
          position={linkPositions[i].p}
          rotY={linkPositions[i].r}
          audio={piece.audio}
          link={piece.link}
          size={piece.size}
          src={piece.src}
/>
```

no need to add another prop to link

File: src/themes/Alto/index.tsx [hunk 3]
```diff
@@ -73,9 +73,10 @@ const Alto = (props: Partial<AltoProps>) => {
         <Grass />
       </Suspense>
       <Sun />
-      <Scrolls count={scrollCount} setCount={setScrollCount} />
+      <Scrolls count={scrollCount} setCount={setScrollCount} />{" "}
```
Comment (alex-shortt) [inline src/themes/Alto/index.tsx:76]: extra {" "}

File: src/themes/Alto/components/Scrolls/Scroll.tsx [hunk 4]
```diff
@@ -96,6 +96,8 @@ const Scroll = (props: JSX.IntrinsicElements["group"] & ScrollProps) => {
         </animated.group>
         <group position-y={0.5} name="innerscroll">
           <animated.group position-x={0.015} scale-y={scale} name="content">
+            {" "}
```
Comment (alex-shortt) [inline src/themes/Alto/components/Scrolls/Scroll.tsx:99]: extra {" "}

File: src/scenes/Silks/modifiers/Spinning.tsx [hunk 5]
```diff
@@ -15,7 +15,7 @@ const Spinning = (props: Props) => {
 
   const group = useRef<Group>();
   const [seed] = useState(Math.random());
-  const limiter = useLimiter(75);
+  const limiter = useLimiter(45);
```
Comment (alex-shortt) [inline src/scenes/Silks/modifiers/Spinning.tsx:18]: i believe we export this from `spacesvr` now

File: src/scenes/Gotham/Balloonski/components/Shop/Kiosk.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  useState,
+} from "react";
+import { Floating } from "spacesvr";
+import Control from "../../../../Silks/components/Kiosk/components/Control";
+import { useBox } from "@react-three/cannon";
+import { Group, Vector3 } from "three";
+import Images from "../../../../Silks/components/Kiosk/components/Images";
+import Description from "../../../../Silks/components/Kiosk/components/Description";
+import { Product, ShopState } from "../../../../Silks/types/shop";
+import { ShopContext } from "../../../../Silks";
```
Comment (alex-shortt) [inline src/scenes/Gotham/Balloonski/components/Shop/Kiosk.tsx:17]: you need to clone the files to keep them separate from the silks files until we can abstract it enough to `spacesvr` or to the `themes/components` folder

File: src/scenes/Gotham/Balloonski/index.tsx [hunk 7]
```diff
@@ -1,21 +1,37 @@
...
+import Kiosks from "./components/Shop/Kiosks";
+import { useShopifyShop } from "../../Silks/utils/shopify";
+
+export const ShopContext = createContext<ShopState>({} as ShopState);
 
 export default function Balloonski() {
+  // const shop = useShopifyShop({
+  //   domain: "silks-by-vp.myshopify.com",
+  //   storefrontAccessToken: "0ee16eee5ad43db15eaf55d74aee5c98",
+  // });
```
Comment (alex-shortt) [inline src/scenes/Gotham/Balloonski/index.tsx:19]: we got the info, i'll dm it to you

File: src/scenes/Gotham/Balloonski/models/Gator.tsx [hunk 8]
```diff
@@ -18,14 +18,14 @@ type GLTFResult = GLTF & {
 };
 
 const FILE_URL =
-  "https://d27rt3a60hh1lx.cloudfront.net/models/gatorhead-1617845675/gatorhead.glb.gz";
+  "https://d27rt3a60hh1lx.cloudfront.net/models/gator-1618894625/gator.glb.gz\n";
```
Comment (alex-shortt) [inline src/scenes/Gotham/Balloonski/models/Gator.tsx:21]: \n

File: src/scenes/Silks/components/Cart/components/Control.tsx [hunk 9]
```diff
@@ -11,8 +11,8 @@ const Control = () => {
   const numY = isMobile ? -1.25 : 6;
 
   const cartStatus =
-    cart && cart.count > 0
-      ? `${cart.count} Item${cart.count > 1 ? "s" : ""}`
+    cart && cart?.count > 0
```
Comment (alex-shortt) [inline src/scenes/Silks/components/Cart/components/Control.tsx:14]: the `cart && ` should check whether cart is undefined, so you shouldn't need the `?` after

File: src/scenes/Silks/components/Cart/index.tsx [hunk 10]
```diff
@@ -40,11 +40,11 @@ const Cart = () => {
...
   });
   useEffect(() => {
-    if (cart.count !== prevCart.current) {
+    if (cart?.count !== prevCart.current) {
       setIncr(incr + 1);
-      prevCart.current = cart.count;
+      prevCart.current = cart?.count;
     }
-  }, [incr, cart.count]);
+  }, [incr, cart?.count]);
```
Comment (alex-shortt) [inline src/scenes/Silks/components/Cart/index.tsx:47]: how come all the `?` pretty sure cart is always defined

**Review by alex-shortt — CHANGES_REQUESTED** [review]
- dm'd you shopify info
- don't share files across scenes folders, you should copy and paste them or abstract them sufficiently to the themes folder or spacesvr

File: src/scenes/Gotham/Balloonski/index.tsx [hunk 11]
```diff
@@ -1,31 +1,56 @@
...
+            name="ghost"
+            position={[-4.5, 1.5, -3.3]}
+            rotation-y={0.67}
+            scale={[10, 10, 10]}
+          />
+        </Floating>
+        <Cart />
+        <Kiosks />
+        <Guide />
+        {/*<Renderer />*/}
```
Comment (alex-shortt) [inline src/scenes/Gotham/Balloonski/index.tsx:52]: remove renderer

File: src/scenes/Gotham/Balloonski/models/Desk.tsx [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+  //   // eslint-disable-next-line react-hooks/rules-of-hooks
+  //   useTrimeshCollision(
+  //     // @ts-ignore
+  //     (hitbox.current.geometry)
+  //       .clone()
+  //       .translate(2, -0.35, 4)
+  //     // .scale(0.05, 0.035, 0.035)
+  //     // .rotateY(-Math.PI/2)
+  //   );
+  // }
```
Comment (alex-shortt) [inline src/scenes/Gotham/Balloonski/models/Desk.tsx:71]: would recommend using `useBox` from `@react-three/cannon`

File: src/themes/components/Artwork.tsx [hunk 13]
```diff
@@ -19,17 +20,20 @@ const Artwork = (props: ArtworkProps) => {
...
+        (piece, i) =>
+          !piece.blank && (
+            <Link
+              position={linkPositions[i].p}
+              rotY={linkPositions[i].r}
+              audio={piece.audio}
+              link={piece.link}
+              size={piece.size}
+              src={piece.src}
+              key={i}
```
Comment (alex-shortt) [inline src/themes/components/Artwork.tsx:33]: sorry why is this blank prop here? feels unnecessary

**Review by alex-shortt — CHANGES_REQUESTED** [review]

File: src/themes/components/Artwork.tsx (hunk 13 above)
Comment (ianyimi) [reply src/themes/components/Artwork.tsx:33]: it's just so I didn't get that react warning in the console

**Review by alex-shortt — APPROVED** [review]

**Pull request merged**